src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -2865,31 +2865,49 @@ private void EnsureForwardTypeToAssemblyMap()
         {
             if (_lazyForwardedTypesToAssemblyMap == null)
             {
-                var typesToAssemblyMap = new Dictionary<string, AssemblyReferenceHandle>();
+                var typesToAssemblyMap = new Dictionary<string, HashSet<AssemblyReferenceHandle>>();
 
                 try
                 {
                     var forwarders = MetadataReader.ExportedTypes;
                     foreach (var handle in forwarders)
                     {
                         ExportedType exportedType = MetadataReader.GetExportedType(handle);
+
                         if (!exportedType.IsForwarder)
                         {
                             continue;
                         }
 
+                        AssemblyReferenceHandle newReference = (AssemblyReferenceHandle)exportedType.Implementation;
+
+                        if (newReference.IsNil)
+                        {
+                            continue;
```

Comment:
Previously, when `(AssemblyReferenceHandle)exportedType.Implementation` was `Nil`, it would still be added to the map. Does this make a difference?